db/migrate/20240601103707_add_polymorphic_association_to_comment.rb

```diff
@@ -0,0 +1,16 @@
+class AddPolymorphicAssociationToComment < ActiveRecord::Migration[7.1]
+  def change
+    remove_index :task_comments, :overseer_assessment_id
+
+    add_column :task_comments, :commentable_type, :string
+    rename_column :task_comments, :overseer_assessment_id, :commentable_id
+
+    TaskComment.find_each do |comment|
+      if comment.commentable_id.present?
+        comment.update(commentable_type: 'OverseerAssessment')
```

Comment:
I think you could then do update all... something like this:

```suggestion
     TaskComment.where('NOT commendable_id IS NULL').in_batches.update_all(commentable_type: 'OverseerAssessment')
```